driver/cam_hal.c

```diff
@@ -289,7 +289,7 @@ static esp_err_t cam_dma_config(const camera_config_t *config)
         cam_obj->frames[x].fb_offset = 0;
         cam_obj->frames[x].en = 0;
         ESP_LOGI(TAG, "Allocating %d Byte frame buffer in %s", alloc_size, _caps & MALLOC_CAP_SPIRAM ? "PSRAM" : "OnBoard RAM");
-        cam_obj->frames[x].fb.buf = (uint8_t *)heap_caps_malloc(alloc_size, _caps);
+        cam_obj->frames[x].fb.buf = (uint8_t *)heap_caps_aligned_alloc(16, alloc_size, _caps);
```

Comment:
This looks reasonable, but I'm afraid there is one caveat.

In IDF releases v4.2 and earlier, memory returned by heap_caps_aligned_calloc _must_ be freed using heap_caps_aligned_free ([docs](https://github.com/espressif/esp-idf/blob/release/v4.2/components/heap/include/esp_heap_caps.h#L137-L138)).

Since this PR changes only the allocation function, the code, as is, may not work correctly in v4.1 and v4.2 releases.

You could modify either to make the allocation function conditional on the IDF version.

Reply:
Thanks, I just build with v4.4, let me test on v4.1 and v4.2